### PR TITLE
Add the required length to create a webservice key into form

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      |  Add the constraints (32 characthers) into the help of the key creation form.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #n/d.
| How to test?      |  See that when you want to create a webservice key, the "key" help will say you the number of needed characthers.
| Possible impacts? | Nothing.

Wording needed:

- Key should be at least %length% characters long. (Admin.Notifications.Info)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27108)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Webservice/WebserviceKeyType.php [hunk 1]
```diff
@@ -87,7 +87,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                'help' => sprintf(
+                    '%s<br>%s',
+                    $this->trans('Webservice account key.', 'Admin.Advparameters.Feature'),
+                    $this->trans(
+                        'Key should be at least %length% characters long.',
+                        'Admin.Notifications.Info',
+                        [
+                            '%length%' => Key::LENGTH,
+                        ]
+                    ),
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Webservice/WebserviceKeyType.php]:
```suggestion
                    )
```

Syntax error :)

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Syntax error with PHP 7.2

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello,

PR is tested and the number of needed characthers in the key help is wel displayed :
image


PR ok also for Samsung Galaxy S7 v6.0 :
image


Many thanks,

**Comment by PierreRambaud** [comment]
Thank you @PrestaEdit

**Comment by PierreRambaud** [comment]
Just notice there were a Waiting for Wording label :roll_eyes: 
We will improve it later, we can do it later :p

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Webservice/WebserviceKeyType.php [hunk 2]
```diff
@@ -87,7 +87,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $builder
             ->add('key', GeneratableTextType::class, [
                 'label' => $this->trans('Key', 'Admin.Advparameters.Feature'),
-                'help' => $this->trans('Webservice account key.', 'Admin.Advparameters.Feature'),
+                'help' => sprintf(
+                    '%s<br>%s',
+                    $this->trans('Webservice account key.', 'Admin.Advparameters.Feature'),
+                    $this->trans(
+                        'Key should be at least %length% characters long.',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Webservice/WebserviceKeyType.php:94]: This is not the right translation domain. It should be Admin.Advparameters.Help

See => https://zeroheight.com/80a6c5a61/v/latest/p/738e72-manage-translation-domains

We also miss the article "The" before "key".

**Pull request merged**